packages/kbn-check-mappings-update-cli/src/check_incompatible_mappings.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { SomeDevLog } from '@kbn/some-dev-log';
+import { Client } from '@elastic/elasticsearch';
+import { createFailError } from '@kbn/dev-cli-errors';
+import type { SavedObjectsTypeMappingDefinitions } from '@kbn/core-saved-objects-base-server-internal';
+
+const TEST_INDEX_NAME = '.kibana_mappings_check';
+
+export async function checkIncompatibleMappings({
+  log,
+  esClient,
+  currentMappings,
+  nextMappings,
+}: {
+  log: SomeDevLog;
+  esClient: Client;
+  currentMappings: SavedObjectsTypeMappingDefinitions;
+  nextMappings: SavedObjectsTypeMappingDefinitions;
+}) {
+  try {
+    log.debug('creating index using current mappings...');
+    await esClient.indices.create({
+      index: TEST_INDEX_NAME,
+      mappings: {
+        dynamic: false,
+        properties: currentMappings,
+      },
+      settings: {
+        mapping: {
+          total_fields: { limit: 1500 },
```

Comment:
NIT: fine for now probably, we may want to import this const from the SO code later to keep the values in sync